### PR TITLE
ci: set up GitHub Actions

### PR DESCRIPTION
**Pull request by vlsi**
The project targets Java 1.5, so Java 8 should be used for the compilation.
However, the tests are run with the current JAVA_HOME, so when running Maven with Java 17 the tests use Java 17.

Sample CI output: https://github.com/vlsi/reload4j/actions/runs/1696556736

### CODE REVIEW

File: src/test/java/org/apache/log4j/util/SunReflectFilter.java [hunk 1]
```diff
@@ -52,6 +52,12 @@ public String filter(String in) {
     if (in.indexOf("at java.lang.reflect.") >= 0) {
       return null;
     }
+    if (in.indexOf("at java.base/jdk.internal.reflect") >= 0) {
+      return null;
+    }
+    if (in.indexOf("at java.base/java.lang.reflect") >= 0) {
+      return null;
+    }
```
Comment (ceki) [inline src/test/java/org/apache/log4j/util/SunReflectFilter.java:60]: SunReflectFilter has been replaced by Log4jAndNothingElseFilter

Comment (vlsi) [reply src/test/java/org/apache/log4j/util/SunReflectFilter.java:60]: This change was still required to make the tests pass

Comment (vlsi) [reply src/test/java/org/apache/log4j/util/SunReflectFilter.java:60]: Ah, I'm wrong I just misjudged the output

**Comment by vlsi** [comment]
The tests pass, and I think it is ready.

**Pull request closed without merging**